includes/db.inc

```diff
@@ -308,6 +311,24 @@ function islandora_solr_get_collection_sort_string($pid, $use_fallbacks = FALSE)
 }
 
 /**
+ * Sets a default of ' asc' if no sort order string is provided.
+ *
+ * @param string $sort_setting
+ *   The sort setting value to apply the default to.
+ */
+function islandora_solr_set_default_collection_sort_order($sort_setting) {
+  $exploded_sort_setting = explode(',', $sort_setting);
+  foreach ($exploded_sort_setting as &$field) {
+    $field = trim($field);
+    if (!preg_match('/\s+(asc|desc)$/', $field)) {
```

Comment:
According to https://cwiki.apache.org/confluence/display/solr/Defining+Fields
> Field names should consist of alphanumeric or underscore characters only and not start with a digit. This is not currently strictly enforced, but other field names will not have first class support from all components and back compatibility is not guaranteed.

So I created some fields
```
<dynamicField name="*_asc" type="string" indexed="true" stored="true" multiValued="false"/>
<dynamicField name="*_desc" type="string" indexed="true" stored="true" multiValued="false"/>
<field name="bob desc" type="string" indexed="true" stored="true" multiValued="false"/>
```

The `_desc` and `_asc` work fine. 

The `bob desc` works to get stuff into solr so I can see the field through the Solr interface. But I couldn't figure out how to query or sort by it in the Solr interface.

Using the collection sort and setting it  to `bob desc` resulted in:
```
string in (bob desc)
field (bob desc)
string out (bob desc)
```

and a Solr error

```
INFO: [collection1] webapp=/solr path=/select params={facet.limit=100&q=RELS_EXT_isMemberOf_uri_ms:("info:fedora/islandora:sp_basic_image_collection"+OR+"islandora:sp_basic_image_collection")+OR+RELS_EXT_isMemberOfCollection_uri_ms:("info:fedora/islandora:sp_basic_image_collection"+OR+"islandora:sp_basic_image_collection")&facet.field=fgs_label_ms&facet.field=RELS_EXT_hasModel_uri_ms&json.nl=map&start=0&facet.mincount=1&fq=-RELS_EXT_isConstituentOf_uri_mt:[*+TO+*]&fq=RELS_EXT_isViewableByUser_literal_ms:"admin"+OR+RELS_EXT_isViewableByRole_literal_ms:"authenticated+user"+OR+RELS_EXT_isViewableByRole_literal_ms:"administrator"+OR+((*:*+-RELS_EXT_isViewableByUser_literal_ms:[*+TO+*])+AND+(*:*+-RELS_EXT_isViewableByRole_literal_ms:[*+TO+*]))&sort=bob+desc&rows=12&facet=true&version=1.2&wt=json} status=400 QTime=1 
Feb 23, 2017 9:34:39 PM org.apache.solr.common.SolrException log
SEVERE: org.apache.solr.common.SolrException: can not sort on multivalued field: bob
	at org.apache.solr.schema.SchemaField.checkSortability(SchemaField.java:170)
	at org.apache.solr.schema.FieldType.getStringSort(FieldType.java:617)
	at org.apache.solr.schema.TextField.getSortField(TextField.java:102)
	at org.apache.solr.schema.SchemaField.getSortField(SchemaField.java:153)
	at org.apache.solr.search.QueryParsing.parseSort(QueryParsing.java:351)
	at org.apache.solr.search.QParser.getSort(QParser.java:281)
	at org.apache.solr.handler.component.QueryComponent.prepare(QueryComponent.java:123)
	at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:187)
	at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:135)
	at org.apache.solr.core.SolrCore.execute(SolrCore.java:1797)
	at org.apache.solr.servlet.SolrDispatchFilter.execute(SolrDispatchFilter.java:637)
	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:343)
	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:141)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:171)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:950)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:408)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1070)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:611)
	at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:314)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
```

Trying `"bob desc"` resulted in 
```
string in ("bob desc")
field ("bob desc")
string out ("bob desc" asc)
```

and the Solr error
```
INFO: [collection1] webapp=/solr path=/select params={facet.limit=100&q=RELS_EXT_isMemberOf_uri_ms:("info:fedora/islandora:sp_basic_image_collection"+OR+"islandora:sp_basic_image_collection")+OR+RELS_EXT_isMemberOfCollection_uri_ms:("info:fedora/islandora:sp_basic_image_collection"+OR+"islandora:sp_basic_image_collection")&facet.field=fgs_label_ms&facet.field=RELS_EXT_hasModel_uri_ms&json.nl=map&start=0&facet.mincount=1&fq=-RELS_EXT_isConstituentOf_uri_mt:[*+TO+*]&fq=RELS_EXT_isViewableByUser_literal_ms:"admin"+OR+RELS_EXT_isViewableByRole_literal_ms:"authenticated+user"+OR+RELS_EXT_isViewableByRole_literal_ms:"administrator"+OR+((*:*+-RELS_EXT_isViewableByUser_literal_ms:[*+TO+*])+AND+(*:*+-RELS_EXT_isViewableByRole_literal_ms:[*+TO+*]))&sort="bob+desc"+asc&rows=12&facet=true&version=1.2&wt=json} status=500 QTime=2 
Feb 23, 2017 9:36:30 PM org.apache.solr.common.SolrException log
SEVERE: null:java.lang.UnsupportedOperationException
	at org.apache.lucene.queries.function.FunctionValues.doubleVal(FunctionValues.java:47)
	at org.apache.lucene.queries.function.ValueSource$ValueSourceComparator.copy(ValueSource.java:168)
	at org.apache.lucene.search.TopFieldCollector$OneComparatorNonScoringCollector.collect(TopFieldCollector.java:86)
	at org.apache.solr.search.DocSetDelegateCollector.collect(DocSetDelegateCollector.java:41)
	at org.apache.lucene.search.BooleanScorer2.score(BooleanScorer2.java:283)
	at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:605)
	at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:297)
	at org.apache.solr.search.SolrIndexSearcher.getDocListAndSetNC(SolrIndexSearcher.java:1553)
	at org.apache.solr.search.SolrIndexSearcher.getDocListC(SolrIndexSearcher.java:1300)
	at org.apache.solr.search.SolrIndexSearcher.search(SolrIndexSearcher.java:395)
	at org.apache.solr.handler.component.QueryComponent.process(QueryComponent.java:410)
	at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:208)
	at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:135)
	at org.apache.solr.core.SolrCore.execute(SolrCore.java:1797)
	at org.apache.solr.servlet.SolrDispatchFilter.execute(SolrDispatchFilter.java:637)
	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:343)
	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:141)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:171)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:950)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:408)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1070)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:611)
	at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:314)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
```

Unless I missed something, I think if someone wants to use a field name with a blank in it they are asking for general trouble.